graphistry/feature_utils.py

```diff
@@ -911,12 +991,19 @@ def process_dirty_dataframes(
         )
         #  now just set the feature names, since dirty cat changes them in
         #  a weird way...
-        data_encoder.get_feature_names_out = callThrough(features_transformed)
-        
-        X_enc = pd.DataFrame(
-            X_enc, columns=features_transformed, index=ndf.index
-        )
-        X_enc = X_enc.fillna(0.0)
+        data_encoder.get_feature_names_out = callThrough(features_transformed) 
+        if 'cudf' not in str(getmodule(ndf)):
+            X_enc = pd.DataFrame(
+                X_enc, columns=features_transformed, index=ndf.index
+            )
+            X_enc = X_enc.fillna(0.0)
+        else:
+            _, _, cudf = lazy_import_has_dependancy_cu_cat()
```

Comment:
(and are there tests for this?)

Reply:
in general you mean? this gpu/feat branch was @tanmoyio attempt to handle this dilema i believe, but i can take it over

i think in future if cudf is passed as general engine it should initiate cucat and we remove feature_engine flag; also e.g. if cucat + umap-learn passed -- just to bypass user selection error / bad combinations

Reply:
Yes Tanmoy is out for a bit unexpectedly, so that would help

And yes we need to handle multiple envs & multiple user prefs in them, so we can do smart defaults + allow overrides . Makes automated testing here important bc so tricky ...

Reply:
there are currently two engine flags -- so full cpu and full gpu work as expected, and independently `feat_eng=dirty_cat` + `umap_eng=cudf` (cpu then gpu)  and `feat_eng=cu_cat` + `umap_eng=umap_learn` (gpu then cpu) work. 

next:
1. add these engine combination tests
2. engine=cudf to encompass cu_cat+cuml or stop worrying and learn to love user prefs